File: app/src/main/java/hpn332/cb/utils/adapter/AdapterListProject.java
package hpn332.cb.utils.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import hpn332.cb.R;
import hpn332.cb.ui.activity.EditActivity;
import hpn332.cb.ui.activity.ListTaskActivity;
import hpn332.cb.utils.Key;
import hpn332.cb.utils.model.ProjectStructure;

public class AdapterListProject extends RecyclerView.Adapter<AdapterListProject.ItemHolder> {

	private LayoutInflater              inflater;
	private ArrayList<ProjectStructure> arrayList;

	public AdapterListProject(Context context, ArrayList<ProjectStructure> arrayList) {

		this.arrayList = arrayList;
		inflater = LayoutInflater.from(context);
	}

	@Override
	public ItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {

		return new ItemHolder(inflater.inflate(R.layout.row_item_project, parent, false));
	}

	@Override
	public void onBindViewHolder(final ItemHolder holder, int position) {holder.onBind(position);}

	@Override
	public int getItemCount() {return arrayList.size();}

	class ItemHolder extends RecyclerView.ViewHolder
			implements View.OnClickListener, View.OnLongClickListener {

		private TextView title, count_task;
		private int position;

		ItemHolder(View view) {
			super(view);

			title = view.findViewById(R.id.title_textView);
			count_task = view.findViewById(R.id.count_task_textView);

			view.setOnClickListener(this);
			view.setOnLongClickListener(this);
		}

		void onBind(int position) {

			this.position = position;

			title.setText(arrayList.get(position).getTitle());
			count_task.setText(arrayList.get(position).getDesc());
		}


		@Override
		public boolean onLongClick(View view) {

			inflater.getContext().startActivity(
					new Intent(inflater.getContext(), EditActivity.class)
							.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
							.putExtra(Key.KEY_TYPE, EditActivity.PROJECT)
							.putExtra(Key.KEY_POSITION, position));

			return true;
		}

		@Override
		public void onClick(View view) {
			inflater.getContext().startActivity(
					new Intent(inflater.getContext(), ListTaskActivity.class)
							.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
							.putExtra(Key.KEY_PROJECT, arrayList.get(position).getId()));
		}
	}
}

File: app/src/main/java/hpn332/cb/view/project/AdapterListProject.kt
package hpn332.cb.view.project

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import hpn332.cb.R
import hpn332.cb.view.task.AListTask
import hpn332.cb.utils.Key
import hpn332.cb.utils.Type
import hpn332.cb.utils.Utils
import hpn332.cb.model.stucture.Project

class AdapterListProject(context: Context) : RecyclerView.Adapter<AdapterListProject.ItemHolder>() {

    private var list: List<Project> = listOf()

    private val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemHolder =
        ItemHolder(inflater.inflate(R.layout.row_item_project, parent, false))

    override fun onBindViewHolder(holder: ItemHolder, position: Int) = holder.onBind()

    override fun getItemCount(): Int = list.size

    fun setData(list: List<Project>) {
        this.list = list
        notifyDataSetChanged()
    }

    inner class ItemHolder(val view: View) : RecyclerView.ViewHolder(view) {

        private val title: TextView = view.findViewById(R.id.title_textView)
        //        private val count_task: TextView = view.findViewById(R.id.count_task_textView)

        fun onBind() {

            val item = list[adapterPosition]

            title.text = item.title
            //            count_task.text = list.desc

            view.setOnClickListener {
                inflater.context.startActivity(
                    Intent(inflater.context, AListTask::class.java)
                        .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                        .putExtra(Key.PROJECT, item.id))
            }

            view.setOnLongClickListener {
                Utils.goTo(inflater.context, Type.EDIT_PROJECT, id = item.id)

                true
            }

        }
    }
}
